lib/sensors/Stemma.cpp

```diff
@@ -0,0 +1,61 @@
+#include "mbed.h"
+#include "I2C.h"
+#include "Stemma.h"
+
+#define I2C_Addr 0x36 << 1                                                //MBED I2C uses 8 bit addressing, so addresses are left shifted by 1 (may need to be shifted by 2)
+
+Stemma::Stemma(bool* initialized){
+    char check[1];
+
+    i2c.write(0x00);                                                        //initialize registers for checking device ID
+    i2c.write(0x01);
+
+    i2c.read(I2C_Addr, check, 1);                                           //read device ID
+
+    if(check[0] != 0x55){                                                   //compare received HW ID Code to correct one
+        *initialized = false;
+    }
+    else{
+        *initialized = true;
+    }
+}
+
+Stemma::~Stemma(){
+}
+
+uint16_t Stemma::Read_Moisture(){
+    char buf[2];
+    uint16_t ret = 65535;
+
+    do{
+        wait_ms(1);
+
+        i2c.write(0x0F);                                                    //initialize registers for reading moisture
+        i2c.write(0x10);
```

Comment:
https://os.mbed.com/docs/mbed-os/v5.15/mbed-os-api-doxy/classmbed_1_1_i2_c.html#aeb2e6757a3348668fa25d43f7932f78e

should probably use this version of the function instead of calling two seperate writes